web3/contract.py

```diff
@@ -432,6 +433,15 @@ def estimateGas(self, transaction=None):
                     "Please ensure that this contract instance has an address."
                 )
 
+        if 'value' in estimate_transaction:
```

Comment:
instead of duplicating this logic in multiple places, I think all of this code can go in the validation middleware.

Reply:
yeah, a helper function would do. 
The problem with the middleware approach is that the `abi` will not be available there.

Reply:
@voith I was wondering the same when I looked at the validation code in middleware since 'abi' is not available there.

@carver Yeah, I will consolidate it.